src/components/Button/Button.tsx

```diff
@@ -0,0 +1,46 @@
+import { ReactNode } from 'react';
+import classNames from 'classnames/bind';
+import { BUTTON_COLOR } from '@/constants/buttonTypes';
+import styles from './Button.module.scss';
+
+type ButtonColorType = (typeof BUTTON_COLOR)[keyof typeof BUTTON_COLOR];
+
+const cn = classNames.bind(styles);
+
+interface ButtonProps {
+  backgroundColor?: ButtonColorType;
+  hoverColor?: ButtonColorType;
+  radius?: 4 | 8;
+  width?: 'parent-full' | 72 | 90 | 120 | 154 | 320;
+  fontSize?: 14 | 18 | 20 | 24;
+  paddingVertical?: 8 | 20;
+  onClick: () => void;
```

Comment:
```suggestion
const BUTTON_COLOR = {
  BACKGROUND_GRAY_40: 'background-gray-40',
  BACKGROUND_PRIMARY: 'background-primary',
  BACKGROUND_PRIMARY_60: 'background-primary-60',
  OUTLINE_GRAY_40: 'outline-gray-40',
  OUTLINE_PRIMARY: 'outline-primary',
  OUTLINE_PRIMARY_60: 'outline-primary-60',
} as const;

type ButtonColorType = (typeof BUTTON_COLOR)[keyof typeof BUTTON_COLOR];

const cn = classNames.bind(styles);

interface ButtonProps {
  backgroundColor?: ButtonColorType;
  hoverColor?: ButtonColorType;
  radius?: 4 | 8;
  width?: 'parent-full' | 72 | 90 | 120 | 154 | 320;
  fontSize?: 14 | 18 | 20 | 24;
  paddingVertical?: 8 | 20;
  onClick: () => void;
  children: ReactNode;
}
```
내부에서 선택해서 사용할 수 있게 만들어도 좋을거같습니다

Reply:
width와 fontSize가 동일한 사이즈로 묶여서 간다면 함꼐 묶어서 사이즈를 특정하셔도 좋을거같아요
ex) `buttonSize: xs | sm | md | lg | xl ;`

Reply:
버튼 탐색해봤는데 width와 fontSize가 동일하지 않은 경우도 있고, width도 타입으로 지정된 것 외에 다른 크기를 가진 버튼들도 있어서 font와 width를 따로 설정했습니다

Reply:
아하 위에 처럼 컴포넌트 안에 객체를 위치 시키는 건 어떻겠냐는 의견이셨군요!! 근데 혹시 저렇게 하면 Button 컴포넌트를 사용할 때는 혹시 어떻게 사용하게 될까용? 

제가 위에서 구현했던건 버튼 사용 시에 아래처럼 사용했는데
`<Button backgroundColor={BUTTON_COLOR.OUTLINE_PRIMARY_60} >`
그럼 background에 직접 클래스명을 작성하도록 하면 조금 더 편할까열!?!

Reply:
props 이름 작성 후 중괄호 안에서 컨트롤 + ㅑ 입력하면 관련 prop 나올걸료? 스토리북에서 확인해도 충분할거라 생각합니다~

Reply:
추가적으로 사이즈 관한내용은 지금도 괜찮지만 한번 디자이너님과 상의해보셔도 좋을거같아요

Reply:
오호 처음 아는 사실이군요! 그럼 그 방식으로 변경해서 수정하겠습니다

Reply:
사이즈 관한 내용은 이미 디자이너님과 상의한 결과로 저렇게 간추려졌습니다!